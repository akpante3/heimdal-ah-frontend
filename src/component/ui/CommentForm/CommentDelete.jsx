import React from 'react';
import FontAwesome from 'react-fontawesome';
import { connect } from 'react-redux';
import { deleteArticleComment } from '../../../actions/CommentActions/CommentActions';
import { deleteCommentReplies } from '../../../actions/ReplyActions/ReplyAction';

/**
  * renderComponent
  * @method Function Function based Component
  * @summary React component for rendering the card components
  * @param {Object} props - React PropTypes
  * @property {String|Object} className - String className compatible object for styling
  * @return {Node} React node containing delete comment and replies
  */
const DeleteButton = (props) => {
  const deleteComment = (e) => {
    e.preventDefault();
    const { commentId, buttonName, replyId } = props;
    if (buttonName === 'comment') {
      props.deleteArticleComment(1, commentId);
    } else {
      props.deleteCommentReplies(commentId, replyId);
    }
  };

  /**
     * Renders the component.
     *
     * @memberof app.components.Comment
     * @return {string} - HTML markup for the component
   */

  return (
    <button className='archive_button btn btn-danger' onClick={e => deleteComment(e)} type='submit'><FontAwesome name='trash' className='' /></button>
  );
};

/**
 * @method module:Reactator.ReduxContainerBuilderMapStateToProps
 *
 * @param {function} mapStateToProps - function mapping redux state to props
 *
 * @return {ReduxContainerBuilder} this builder
*/
const mapStateToProps = state => ({
  comment: state.comment,
});

export default connect(
  mapStateToProps,
  {
    deleteArticleComment, deleteCommentReplies,
  },
)(DeleteButton);
