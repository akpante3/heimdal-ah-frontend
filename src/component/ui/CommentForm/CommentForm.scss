@import '../../../styles/variables.scss';

.form__comment {
  font-family:$font_family_reem-kufi;
  border-top: 1px solid $heimdal-comment-border;
  
  .comment__head__a {
    font-size: 25px;
    margin-top: 60px;
    padding-top: 70px;
  }
  .comment__head__b {
    font-size: 20px;
  }
  .comment-form {
    border: 1px solid $heimdal-comment-border;
    border-radius: 0!important; 
    margin-right: 70px!important;
    width: 100%;
  }
  
  .flex-container__a {
    display: flex;
    justify-content: flex-end;
    margin-top: 10px!important;
  }
  .flex-container__b {
    display: flex;
    justify-content: space-between;
  }
  
  .comment_check {
    margin-right: 30px!important;
    margin-top: 8px;

  }

  .private__comment {
    margin-right: 50px;
    margin-top: 10px;
    font-size: 15px;
    color: $heimdal-private-comment-color;
    font-family: $font-family__cardo;
  }
}

