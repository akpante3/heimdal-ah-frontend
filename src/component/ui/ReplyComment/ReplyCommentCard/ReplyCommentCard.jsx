import React from 'react';
import FontAwesome from 'react-fontawesome';
import { connect } from 'react-redux';
import DeleteButton from '../../CommentForm/CommentDelete';
import './ReplyCommentCard.scss';

const ReplyCommentCard = (props) => {
  const { user } = props.auth;
  let replies = props.replies[props.commentId];
  replies = replies || [];
  const renderCard = () => replies.map(item => (
    <section key={item.id} className="comment-section container">
      <div className="comment-card row">
        <div className="comment-image col-md-2 col-sm-2">
          <img className="comment-card-img" src={item.profile.image} alt="user-comment-logo" />
        </div>
        <div className="comment-card-body col-md-9 col-sm-9">
          <div className="comment-body-top">
            <span className="reply-name">{item.profile.username}</span>
            <span className="comment-date small_text">
              <FontAwesome name="clock" className="fav_icons" />
            </span>
            <span className="small_text">
              <FontAwesome name="calendar" className="fav_icons" />
                09: 00 am
            </span>
            { item.userId === user.userId
              ? <DeleteButton replyId={item.id} buttonName='reply' commentId={props.commentId} />
              : <div />
            }
          </div>
          <div className="reply-body">{item.content}</div>
          <div className="comment-body-footer">
            <span />
            <span />
          </div>
        </div>
      </div>
    </section>
  ));

  return <div>{replies || Object.keys(replies).length !== 0 ? renderCard() : <div />}</div>;
};

const mapStateToProps = state => ({
  auth: state.auth,
  replies: state.replies,
});

export default connect(
  mapStateToProps,
  {},
)(ReplyCommentCard);
