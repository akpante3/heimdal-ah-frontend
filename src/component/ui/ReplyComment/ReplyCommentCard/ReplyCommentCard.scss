.reply-body {
  margin-left: 30px;
}
.reply-name {
  margin-left: 40px;
}

.small_text {
  margin-left: 20px;
}