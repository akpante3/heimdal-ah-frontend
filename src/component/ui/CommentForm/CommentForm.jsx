/* eslint-disable react/no-unused-prop-types */
/* eslint-disable react/forbid-prop-types */
import React, { Component } from 'react';
import { FormGroup, FormControl } from 'react-bootstrap';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Button from '../Buttons/Button';
import { postArticleComment } from '../../../actions/CommentActions/CommentActions';
import Alert from '../Alert/Alert';
import './CommentForm.scss';

/**
  * renderComponent
  * @method Class Class based Component
  * @summary React component for rendering the card components
  * @param {object}  comment contains the comments of an article
  * @param {Object} props - React PropTypes
  * @property {String|Object} className - String className compatible object for styling
  * @return {Node} React node containing comment form view
  */
export class CommentForm extends Component {
  state = {
    formIsValid: false,
    commentError: '',
    content: '',
    checkbox: false,
  }

   onChange = (e) => {
     this.setState({
       [e.target.name]: e.target.value,
       commentError: '',
     });
   }

   onBlur = (e) => {
     if (e.target.value === '') {
       this.setState({
         commentError: 'Content cannot be Empty',
       });
     } else {
       this.setState({
         formIsValid: true,
         commentError: '',
       });
     }
   }

   onCheck = () => {
     const { checkbox } = this.state;
     this.setState({
       checkbox: !checkbox,
       commentError: '',
     });
   }

   submitForm = (e) => {
     e.preventDefault();
     const { user } = this.props.auth;
     const { formIsValid, content, checkbox } = this.state;
     if (content.length >= 1500) {
       this.setState({
         formIsValid: false,
         commentError: "can't have more than 1500 characters",
       });
     } else if (formIsValid) {
       const data = {
         content,
         isPrivate: checkbox,
       };

       this.props.postArticleComment(this.props.articleId, data, user.image);
       this.setState({
         content: '',
       });
     } else {
       this.setState({
         commentError: 'Content cannot be Empty',
       });
     }
   }

   /**
     * Renders the component.
     *
     * @memberof app.components.Comment
     * @return {string} - HTML markup for the component
   */
   render() {
     const { checkbox, commentError } = this.state;
     return (
       <div className='container form__comment'>
         <div className='comment__head'>
           <p className='comment__head__a'>ADD COMMENT</p>
           <p className='comment__head__b'>Comment</p>
         </div>
         <div>
           {' '}
           {commentError !== '' && <Alert type='warning' message={this.state.commentError} title='Invalid Credenntials' /> }
           <div />
         </div>
         <form className='' onSubmit={this.submitForm}>
           <FormGroup controlId="formControlsTextarea">
             <FormControl
               onBlur={this.onBlur}
               onChange={this.onChange}
               componentClass="textarea"
               value={this.state.content}
               name='content'
               className='comment-form'
               rows="7"
               cols="70"
               placeholder="textarea"
             />
           </FormGroup>
           <div className='flex-container__a'>
             <div className="form-check flex-container__b">
               <div className='private__comment'>Make it Private</div>
               <div className='comment_check'>
                 <input type="checkbox" name='checkbox' onChange={e => this.onCheck(e)} className="form-check" value={!checkbox} />
               </div>
             </div>
             <Button type="comment" className='comment-button' label="Add Comment" />
           </div>
         </form>
       </div>
     );
   }
}

CommentForm.propTypes = {
  postArticleComment: PropTypes.func.isRequired,
  comment: PropTypes.object.isRequired,
  auth: PropTypes.object.isRequired,
  error: PropTypes.object.isRequired,
};

/**
 * @method module:Reactator.ReduxContainerBuilderMapStateToProps
 *
 * @param {function} mapStateToProps - function mapping redux state to props
 *
 * @return {ReduxContainerBuilder} this builder
*/
const mapStateToProps = state => ({
  comment: state.comment,
  auth: state.auth,
  error: state.errors,
});

export default connect(mapStateToProps, { postArticleComment })(CommentForm);
