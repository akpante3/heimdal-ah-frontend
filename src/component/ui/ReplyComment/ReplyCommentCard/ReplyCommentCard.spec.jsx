/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import { shallow } from 'enzyme';
import { ReplyCommentCard } from './ReplyCommentCard';
import { DeleteButton } from '../../CommentForm/CommentDelete';

describe('rendering', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      auth: '',
      replies: '',
    };
  });

  it('renders One Comment Card components', () => {
    wrapper = shallow(<ReplyCommentCard {...props} />);
    expect(wrapper.find(DeleteButton)).toEqual({});
  });

  it('should render the component', () => {
    wrapper = shallow(<ReplyCommentCard {...props} />);
    expect(wrapper.length).toBe(1);
  });

  it('should test snapshot', () => {
    wrapper = shallow(<ReplyCommentCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
