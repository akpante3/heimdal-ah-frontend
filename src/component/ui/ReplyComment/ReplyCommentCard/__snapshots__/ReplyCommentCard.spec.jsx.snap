// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should test snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReplyCommentCard
    auth=""
    replies=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-md-12 text-center"
        >
          
        </div>,
        <div />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "col-md-12 text-center",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-md-12 text-center"
          >
            
          </div>,
          <div />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "col-md-12 text-center",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
